web/packages/teleport/src/Integrations/Enroll/Shared/Tile.tsx

```diff
@@ -0,0 +1,307 @@
+/**
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ReactNode } from 'react';
+import { Link as InternalLink } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { Link as ExternalLink, Flex, Label, Text } from 'design';
+import { NewTab } from 'design/Icon';
+import * as Icons from 'design/Icon';
+import { ResourceIconName } from 'design/ResourceIcon/resourceIconSpecs';
+
+import cfg from 'teleport/config';
+import { IntegrationKind } from 'teleport/services/integrations';
+
+import { IntegrationIcon } from '../common';
+import { type IntegrationTileSpec } from '../IntegrationTiles/integrations';
+import {
+  GenericNoPermBadge,
+  renderExternalAuditStorageBadge,
+} from '../IntegrationTiles/IntegrationTiles';
+import { integrationTagOptions, type IntegrationTag } from './common';
+
+type IntegrationLink = {
+  url: string | undefined;
+  external?: boolean;
+  onClick?: () => void;
+};
+
+export function IntegrationTileWithSpec({
+  spec,
+  hasIntegrationAccess = true,
+  hasExternalAuditStorage = true,
+}: {
+  spec: IntegrationTileSpec;
+  hasIntegrationAccess?: boolean;
+  hasExternalAuditStorage?: boolean;
+}) {
+  const link = hasExternalAuditStorage
+    ? { external: false, url: cfg.getIntegrationEnrollRoute(spec.kind) }
+    : null;
+
+  const hasAccess = (() => {
+    if (spec.kind === IntegrationKind.ExternalAuditStorage) {
+      const externalAuditStorageEnabled =
+        cfg.entitlements.ExternalAuditStorage.enabled;
+      return (
+        hasIntegrationAccess &&
+        hasExternalAuditStorage &&
+        externalAuditStorageEnabled
+      );
+    }
+
+    return hasIntegrationAccess;
+  })();
```

Comment:
I'm not against IIFE, but it makes reading code a bit harder as indentation increases, which makes it more difficult to see where the function returns or ends.

WDYT of something like this:
```tsx
let hasAccess = hasIntegrationAccess;
if (spec.kind === IntegrationKind.ExternalAuditStorage) {
  hasAccess &&=
    cfg.entitlements.ExternalAuditStorage.enabled && hasExternalAuditStorage;
}
```